config/lib/logger.js

```diff
@@ -0,0 +1,80 @@
+﻿'use strict';
+
+var _ = require('lodash'),
+  config = require('../config'),
+  chalk = require('chalk'),
+  fs = require('fs');
+
+// list of valid formats for the logging
+var validFormats = ['combined', 'common', 'dev', 'short', 'tiny'];
+
+// build logger service
+var logger = {
+  getFormat: getLogFormat, // log format to use
+  getOptions: getLogOptions // log options to use
+};
+
+// export the logger service
+module.exports = logger;
+
+/**
+ * The format to use with the logger
+ *
+ * Returns the log.format option set in the current environment configuration
+ */
+function getLogFormat () {
+  var format = config.log && config.log.format ? config.log.format.toString() : 'combined';
+
+  // make sure we have a valid format
+  if (!_.includes(validFormats, format)) {
+    format = 'combined';
+
+    if (process.env.NODE_ENV !== 'test') {
+      console.log();
+      console.log(chalk.yellow('Warning: An invalid format was provided. The logger will use the default format of "' + format + '"'));
+      console.log();
+    }
+  }
+
+  return format;
+}
+
+/**
+ * The options to use with the logger
+ *
+ * Returns the log.options object set in the current environment configuration.
+ * NOTE: Any options, requiring special handling (e.g. 'stream'), that encounter an error will be removed from the options.
+ */
+function getLogOptions () {
+  var options = config.log && config.log.options ? _.clone(config.log.options, true) : {};
+
+  // check if the current environment config has the log stream option set
+  if (_.has(options, 'stream')) {
+
+    try {
+
+      // create the WriteStream to use for the logs
+      if (options.stream.length) {
+        options.stream = fs.createWriteStream(process.cwd() + '/' + config.log.options.stream, { flags: 'a' });
```

Comment:
@mleanos, great job on this!

I think it would be more robust to provide an ability to specify the full path to the directory which we would like logs to be stored. For example, I like my logs to go to `/var/log/meanjs`. Your code assumes the root directory if the cwd.